intl.emu

```diff
@@ -142,57 +138,75 @@
+          1. Let _m_ be the String consisting of the digits of the decimal representation of _n_ (in order, with no leading zeroes).
+        1. If _e_ &ge; (_p_ - 1), then
           1. Set _m_ to the string-concatenation of _m_ and _e_ - _p_ + 1 occurrences of the code unit 0x0030 (DIGIT ZERO).
           1. Let _int_ be _e_ + 1.
-        1. Else if _e_ ≥ 0, then
+        1. Else if _e_ &ge; 0, then
           1. Set _m_ to the string-concatenation of the first _e_ + 1 code units of _m_, the code unit 0x002E (FULL STOP), and the remaining _p_ - (_e_ + 1) code units of _m_.
           1. Let _int_ be _e_ + 1.
         1. Else,
-          1. Assert: _e_ < 0.
+          1. Assert: _e_ &lt; 0.
           1. Set _m_ to the string-concatenation of *"0."*, -(_e_ + 1) occurrences of the code unit 0x0030 (DIGIT ZERO), and _m_.
           1. Let _int_ be 1.
-        1. If _m_ contains the code unit 0x002E (FULL STOP) and _maxPrecision_ > _minPrecision_, then
+        1. If _m_ contains the code unit 0x002E (FULL STOP) and _maxPrecision_ &gt; _minPrecision_, then
           1. Let _cut_ be _maxPrecision_ - _minPrecision_.
-          1. Repeat, while _cut_ > 0 and the last code unit of _m_ is 0x0030 (DIGIT ZERO),
+          1. Repeat, while _cut_ &gt; 0 and the last code unit of _m_ is 0x0030 (DIGIT ZERO),
             1. Remove the last code unit from _m_.
             1. Set _cut_ to _cut_ - 1.
           1. If the last code unit of _m_ is 0x002E (FULL STOP), then
             1. Remove the last code unit from _m_.
         1. Return the Record { [[FormattedString]]: _m_, [[RoundedNumber]]: _xFinal_, [[IntegerDigitsCount]]: _int_, [[RoundingMagnitude]]: _e_–_p_+1 }.
       </emu-alg>
+
+      <emu-clause id="sec-decimal128torawprecision" type="abstract operation">
+        <h1>
+          Decimal128ToRawPrecision (
+            _x_: a finite Decimal128 value for which sign(_x_) = 1,
+            _minPrecision_: an integer in the inclusive interval from 1 to 21,
+            _maxPrecision_: an integer in the inclusive interval from 1 to 21,
+            _unsignedRoundingMode_: a specification type from the Unsigned Rounding Mode column of <emu-xref href="#table-intl-unsigned-rounding-modes"></emu-xref>, or *undefined*,
+          ): a Record with fields [[FormattedString]] (a String), [[RoundedNumber]] (a mathematical value), [[IntegerDigitsCount]] (an integer), and [[RoundingMagnitude]] (an integer)
+        </h1>
+        <dl class="header">
+          <dt>description</dt>
+          <dd>
+            <p>It is the equivalent of <emu-xref href="#sec-torawprecision">ToRawPrecision</emu-xref> for Decimal128 values.</p>
+          </dd>
+        </dl>
+        <emu-alg>
+          1. Let _v_ be cohort(_x_).
+          1. Let _valueToProcess_ be _v_.
+          1. If _v_ is *+0*<sub>𝔻</sub>, set _valueToProcess_ to 0.
+          1. Otherwise, return ToRawPrecision(_valueToProcess_, _minPrecision_, _maxPrecision_, _unsignedRoundingMode_).
```

Comment:
Issue: You should at some point here pull the precision from the Decimal to retain trailing zeros.

Reply:
That should be taken care of now -- we serialize a Decimal128 value now using the Decimal128ToDecimalString AO, turning off canonicalization. This gives us the raw underlying data, including any trailing zeroes.